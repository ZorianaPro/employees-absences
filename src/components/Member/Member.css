.Member {
  display: flex;
}

.Member .Member-Avatar {
  width: 3em;
  height: 3em;
  border-radius: 50%;
  position: relative;
}

.Member .Member-Avatar > span {
  font-family: var(--fontDancing);
  font-size: var(--fontSize24);
  color: var(--colorRaisinBlack);
  position: absolute;
  top: 50%;
  left: 50%;
  transform: translate(-50%, -50%);
}