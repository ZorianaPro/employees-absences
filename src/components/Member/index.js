import React from 'react';
import './Member.css';

const colors = [
  '#6EA4BF',
  '#AA4465',
  '#FFF05A',
  '#FFCCC9',
  '#C1EEFF',
  '#81AE9D',
  '#63B0CD',
  '#90F1EF',
  '#B4D6D3',
  '#81B29A',
  '#6EA4BF',
  '#AA4465',
  '#FFF05A',
  '#FFCCC9',
  '#C1EEFF',
  '#81AE9D',
  '#63B0CD',
  '#90F1EF',
  '#B4D6D3',
  '#81B29A'
];

const Member = ({
  id,
  userId,
  name
}) => (
  <div className="Member"
    key={ `user-${userId}` }>
    <div className="Member-Avatar"
      style={{ backgroundColor: colors[Math.min(id, 10)] }}>
      <span>{ name[0] }</span>
    </div>
    <p>{ name }</p>
  </div>
);

export default Member;