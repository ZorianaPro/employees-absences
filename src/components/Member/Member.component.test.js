import React from 'react';
import Member from './';
import { shallow, mount } from 'enzyme';
const data = {
  'id': 1,
  'userId': 664,
  'name': 'Zozo'
};

describe('Member', () => {
  it('renders without crashing', () => {
    expect(shallow.bind(
      shallow,
      <Member
        id={ data.id }
        userId={ data.userId }
        name={ data.name }
      />
    )).not.toThrow();
  });

  it('renders as expected', () => {
    expect(mount(
      <Member
        id={ data.id }
        userId={ data.userId }
        name={ data.name }
      />
    )).toMatchSnapshot();
  });

  describe('name', () =>  {
    let member;
    beforeEach(() => {
      member = mount(
        <Member
          id={ data.id }
          userId={ data.userId }
          name={ data.name }
        />
      );
    });

    it('renders name', () => {
      expect(member.find('div p').text())
        .toEqual('Zozo');
    });
    it('renders avatar with first name character', () => {
      expect(member.find('.Member-Avatar span').text())
        .toEqual('Z');
    });
  });
});