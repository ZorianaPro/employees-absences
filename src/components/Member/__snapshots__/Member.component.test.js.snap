// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member renders as expected 1`] = `
<Member
  id={1}
  name="Zozo"
  userId={664}
>
  <div
    className="Member"
    key="user-664"
  >
    <div
      className="Member-Avatar"
      style={
        Object {
          "backgroundColor": "#AA4465",
        }
      }
    >
      <span>
        Z
      </span>
    </div>
    <p>
      Zozo
    </p>
  </div>
</Member>
`;
